backend/src/main/java/gov/cdc/usds/simplereport/service/OktaService.java

```diff
@@ -0,0 +1,121 @@
+package gov.cdc.usds.simplereport.service;
+
+import org.springframework.stereotype.Service;
+
+import java.util.Map; 
+import java.util.HashMap;
+
+import org.springframework.beans.factory.annotation.Value;
+
+import com.okta.sdk.client.Client;
+import com.okta.sdk.client.Clients;
+import com.okta.sdk.resource.user.User;
+import com.okta.sdk.resource.user.UserBuilder;
+import com.okta.sdk.resource.group.GroupList;
+import com.okta.sdk.resource.group.GroupType;
+import com.okta.sdk.resource.group.GroupBuilder;
+import com.okta.sdk.authc.credentials.TokenClientCredentials;
+
+import gov.cdc.usds.simplereport.db.model.ApiUser;
+
+import gov.cdc.usds.simplereport.config.authorization.OrganizationRole;
+
+/**
+ * Handles all user/organization management in Okta
+ */
+@Service
+public class OktaService {
+
+    @Value("${okta.entity-mgmt.org-url}")
+    private String ORG_URL;
+    @Value("${OKTA_API_KEY}")
+    private String API_TOKEN;
+    @Value("${simple-report.authorization.role-prefix}")
+    private String ROLE_PREFIX;
+
+    Client _client;
+
+
+    public OktaService() {
+        _client = Clients.builder()
+            .setOrgUrl(ORG_URL) 
+            .setClientCredentials(new TokenClientCredentials(API_TOKEN))
+            .build();
+    }
+
+    public void createUser(ApiUser apiUser) {
```

Comment:
PersonName (ie nameInfo) is also part of the DB model. is that an issue?

Reply:
Sorry, I didn't mean the model, I meant the entity: NameInfo is a wrapper around some abstract name information, ApiUser is a specific row in the database. My theory was that we should be able to do the Okta call based on the information Okta needs, which is the e-mail and name, but not the internal ID etc.

Reply:
Actually, the required input looks a lot like the IdentityAttributes object, no?

Reply:
just to spell it out explicitly, are you suggesting createUser(firstName, lastName, etc...) ? Or should I sub in a new Okta-specific NameInfo-esque class? (i'm assuming the former.)

Reply:
IdentityAttributes is a service model class that extends PersonName (the DB model class) 😆 let me know what you feel is best here.

Reply:
In that IdentityAttributes is the purpose-built model for holding exactly this information, it's probably the right one to use.

I feel like we're again slightly at cross-purposes here--my concern over using an ApiModel as an input to a service call is not that it comes from the `db` package, it's that it couples the Okta service to the database layer unnecessarily. The auxiliary models and service models are just data containers that we created to avoid having everything be a list of strings. 🙂